django/forms/widgets.py

```diff
@@ -48,15 +48,32 @@ def __init__(self, media=None, css=None, js=None):
                 css = {}
             if js is None:
                 js = []
-        self._css = css
-        self._js = js
+        self._css_lists = [css]
+        self._js_lists = [js]
 
     def __repr__(self):
         return 'Media(css=%r, js=%r)' % (self._css, self._js)
 
     def __str__(self):
         return self.render()
 
+    @property
+    def _css(self):
+        css = self._css_lists[0]
+        for obj in filter(None, self._css_lists[1:]):
+            css = {
+                medium: self.merge(css.get(medium, []), obj.get(medium, []))
+                for medium in css.keys() | obj.keys()
```

Comment:
Both `keys()` calls are unnecessary, `dict.__iter__` yields keys. Also not sure why the `filter(None, ...)` is required.

Reply:
Iterating over a dictionary's keys works, but creating a union does not without the `dict_keys` object:

 ```
>>> a = {"a": 1}
>>> b = {"b": 2}
>>> list(a|b)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for |: 'dict' and 'dict'
>>> list(a.keys()|b.keys())
['b', 'a']
```